### DISCUSSION TITLE
Help:    missing library gdiplus.dll (system.drawing.common)  with .net core 5 base image mcr.microsoft.com/dotnet/aspnet:5.0

### DISCUSSION THREAD

**Discussion by llyons** [dotnet/dotnet-docker#2862] [General]
Hi,

we have the problem with a missing library gdiplus.dll (system.drawing.common)  with .net core 5 base image mcr.microsoft.com/dotnet/aspnet:5.0.   In the past we did use server core images but these where too bulky at 7 gb per image.  

We are constrained to using Windows containers at the moment.   

This is related to issue  https://github.com/dotnet/dotnet-docker/issues/1098

Is there any solution to this?  

At the moment I have asked our developers to look for other ways to handle the functions done with system.drawing.common as a parallel task.

The container / app solution we have today completely works in every other way except the GDI library stuff.

**Comment by llyons** [comment]
also should i be asking this question in a different area?

**Reply by MichaelSimons** [reply]
What is your Windows host version?  If it's LTSC you could make use of the Windows Server Core images the .NET team provides e.g. mcr.microsoft.com/dotnet/aspnet:5.0-windowsservercore-ltsc2019.  See https://github.com/dotnet/dotnet-docker/issues/2377 for details on the supported Windows Server Core images.  

Regarding the size, the .NET team is limited by the size of the Windows base images.  gdiplus.dll requires Windows Server Core or the full sku.

**Reply by llyons** [reply]
Thanks for the response.  I have tried the windows server core and the images for the container are very large (6-7 gb) instead of 300-400mb.  

Do you know if libaries like the gdiplus.dll and gdi32.dll etc can be placed on the light weight base image manually and installed into the gac using gacutil?  

Is this possible?   

we are also looking at using the buster-slim images but running into an error on all outbound calls to api endpoints from on the container itself.

WHAT:Get() service call failed!
System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
 ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception.
 ---> Interop+OpenSsl+SslException: SSL Handshake failed with OpenSSL error - SSL_ERROR_SSL.
 ---> Interop+Crypto+OpenSslCryptographicException: error:141A318A:SSL routines:tls_process_ske_dhe:dh key too small

error:141A318A:SSL routines:tls_process_ske_dhe:dh key too small
